### ISSUE TITLE
Stackdriver output refreshes oauth2 token unnecessarily often

### ISSUE THREAD

**Issue by ThommyH** [fluent/fluent-bit#3324]
## Bug Report

**Describe the bug**
After the fix which made stackdriver plugin refresh token periodically, we see the issue, that tokens are refreshed too often (every 5 seconds). This is probably overkill and according to the code [here](https://github.com/fluent/fluent-bit/blob/v1.7.3/plugins/out_stackdriver/stackdriver.h#L31) it should be every 50 minutes. I would argue further and suggest to respect the expiration time of the token in the calculation instead of hardcoding a long time.

**To Reproduce**
Setup stackdriver output plugin. Let it run for an hour and see logs.

**Expected behavior**
Less frequent token refreshes.

**Logs**
```
[2021/04/06 13:25:19] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:19] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:19] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:19] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:19] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:19] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:19] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:19] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:19] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:19] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:19] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:19] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:24] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:24] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:24] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:24] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:24] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:24] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/04/06 13:25:24] [ info] [oauth2] HTTP Status=200
[2021/04/06 13:25:24] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
```

**Comment by sharkymcdongles** [comment]
@hsmatulisgoogle

**Comment by ThommyH** [comment]
Any updates on this one?

**Comment by sharkymcdongles** [comment]
Someone created a pull request from the old ticket:

https://github.com/fluent/fluent-bit/pull/3373

**Comment by JeffLuoo** [comment]
I tested with #3373 and still saw the same error
```
[2021/05/06 12:44:14] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:14] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:14] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:15] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:15] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:15] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:15] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:15] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:15] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:15] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:15] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:15] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:15] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:16] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:16] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:17] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:17] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:17] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:17] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:18] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:18] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:18] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:18] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:18] [ info] [oauth2] HTTP Status=200
[2021/05/06 12:44:18] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2021/05/06 12:44:19] [ info] [oauth2] HTTP Status=200
```

@sharkymcdongles 
Did you get this problem resolved with #3373?
